### ISSUE TITLE
Look up values in plan_hierarchy with `bolt lookup`

### ISSUE THREAD

**Issue by beechtom** [puppetlabs/bolt#2815]
## Use Case
The `bolt lookup` command is helpful for looking up values using the `hierarchy` key. It would also be helpful to permit lookups using the `plan_hierarchy` key, especially for cases where there are many files in the hierarchy or a user uses interpolations (even though interpolations here are not officially supported).

## Describe the Solution You Would Like
A command-line option could be specified to tell Bolt to use the `plan_hierarchy` when looking up values. For clarity the command-line option could be `--plan-hierarchy`, or for simplicity it could be `--plan`.

```shell
bolt lookup <key> --plan-hierarchy
```

It would also be helpful to provide variables for interpolation on the command-line. Syntax could be similar to providing params to tasks and plans, using VAR=VAL syntax.

```shell
bolt lookup <key> --plan-hierarchy foo=bar baz=bak
```
Include a note in the docs about how plan_hierarchy interpolations can be counterintuitive because they may happen in a different scope than they expect, and we may change the scope they are evaluated in further down the line. For this command the variables are being provided so the scoping / expected value won't change. Maybe a bullet list of "expectations" on the "Using Bolt with Hiera" doc.

We should also add support for passing variable values to the lookup command for the hierarchy, which should have the same precedence as plan variables with target variables for the regular hierarchy lookup.

## Questions
- **What happens when the hiera config does not include a `plan_hierarchy` key?**
  Maintain same behavior as plans: fall back to the `hierarchy` key. This will cause hierarchies that use fact interpolation to break.

**Comment by beechtom** [comment]
Closed by #2589

**Comment by beechtom** [comment]
Closed by #2589
